### PR TITLE
Add s3 readme

### PR DESCRIPTION
**Pull request by jkerr321**
# Description

Add README for the upload-assets-to-s3 plugin

# Checklist:

- [x] My branch has been rebased onto the latest commit on main (don't merge main into your branch)
- [x] My commit messages are [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/), for example: `feat(circleci): add support for nightly workflows`, `fix: set Heroku app name for staging apps too`

### CODE REVIEW

File: plugins/upload-assets-to-s3/README.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+## Testing uploads using the review bucket
+
+You can test uploads to S3 locally on your review bucket to check that you are happy with the configuration. To do this set your `NODE_ENV` to `branch`:
+
+```bash
+$ export NODE_ENV=branch
+```
+
+If the AWS keys names for accessing the review bucket are different to the prod bucket then update the those in the `.toolkitrc.yml`.
```
Comment (ivomurrell) [inline plugins/upload-assets-to-s3/README.md:59]: **todo**: typo
```suggestion
If the AWS key names for accessing the review bucket are different to the prod bucket then update to those in the `.toolkitrc.yml`.
```

File: plugins/upload-assets-to-s3/README.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+plugins:
+	- '@dotcom-tool-kit/upload-assets-to-s3'
+```
+
+Install this plugin's tasks:
+
+```sh
+npx dotcom-tool-kit --install
+```
+
```
Comment (ivomurrell) [inline plugins/upload-assets-to-s3/README.md:25]: **issue**: this plugin doesn't have add any hooks so we don't need to install anything new

```suggestion
```

File: plugins/upload-assets-to-s3/README.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+## Options
+| Key | Description | Default value |
+|-|-|-|
+| `accessKeyIdEnvVar` | variable name of the project's aws access key id | no default value - for backwards compatability the plugin falls back to the default value for `accessKeyId` |
+| `secretAccessKeyEnvVar` | variable name of the project's aws secret access key | no default value - for backwards compatability the plugin falls back to the default value for `secretAccessKey` |
+| `accessKeyId` | **DEPRECATED** variable name of the project's aws access key id | 'aws_access_hashed_assets' |
+| `secretAccessKey` | **DEPRECATED** variable name of the project's aws secret access key | 'aws_secret_hashed_assets' |
+| `directory` | the folder in the project whose contents will be uploaded to S3 | 'public' |
+| `reviewBucket` | the development or test S3 bucket | ['ft-next-hashed-assets-preview'] |
+| `prodBucket` | production S3 bucket/s; an array of strings | ['ft-next-hashed-assets-prod', 'ft-next-hashed-assets-prod-us'] |
```
Comment (ivomurrell) [inline plugins/upload-assets-to-s3/README.md:35]: **question**: if you do pass an array of string does this mean the same files will be uploaded in the same places, just in different buckets?

File: plugins/upload-assets-to-s3/README.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+'@dotcom-tool-kit/upload-assets-to-s3':
+  '@dotcom-tool-kit/upload-assets-to-s3':
+    accessKeyId: AWS_ACCESS
+    secretAccessKey: AWS_KEY
+    prodBucket: ['ft-next-service-registry-prod', 'ft-next-service-registry-prod-us']
+    reviewBucket: ['ft-next-service-registry-dev']
+    destination: ''
+```
+
+## Testing uploads using the review bucket
```
Comment (ivomurrell) [inline plugins/upload-assets-to-s3/README.md:51]: **praise**: it's really useful to have a section like this, taken from your own experience!

File: plugins/upload-assets-to-s3/README.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    - UploadAssetsToS3
+```
+
+Then running `npm run build` will run the `UploadAssetsToS3` task on your review bucket.
+
+## Tasks
+
+| Name | Description | Preconfigured Hook|
+|-|-|-|
+| UploadAssetsTos3 | Uploads provided files to a given S3 bucket | release:remote |
```
Comment (ivomurrell) [inline plugins/upload-assets-to-s3/README.md:79]: **nitpick**: formatting, typo

```suggestion
| `UploadAssetsToS3` | Uploads provided files to a given S3 bucket | `release:remote` |
```

File: plugins/upload-assets-to-s3/README.md (hunk 1 above)
Comment (jkerr321) [reply plugins/upload-assets-to-s3/README.md:59]: Thanks!

File: plugins/upload-assets-to-s3/README.md (hunk 4 above)
Comment (jkerr321) [reply plugins/upload-assets-to-s3/README.md:51]: Well I actually stole some of this from your Cypress README so I'll throw that praise right back at you!

File: plugins/upload-assets-to-s3/README.md (hunk 3 above)
Comment (jkerr321) [reply plugins/upload-assets-to-s3/README.md:35]: [Yes](https://github.com/Financial-Times/dotcom-tool-kit/blob/main/plugins/upload-assets-to-s3/src/tasks/upload-assets-to-s3.ts#L56-L67):

```ts
        for (const bucket of bucketByEnv) {
          const params = {
            Bucket: bucket,
            Key: key,
            Body: body,
            ACL: 'public-read',
            ContentType: `${type}; charset=utf-8`,
            ContentEncoding: encoding,
            CacheControl: options.cacheControl
          }
          currentBucket = params.Bucket
          const data = await s3.upload(params).promise()
```

Comment (jkerr321) [reply plugins/upload-assets-to-s3/README.md:35]: I have clarified in the README 😊

**Review by ivomurrell — APPROVED** [review]
looks goooooood

**Pull request merged**